lib/web_ui/lib/src/engine/canvaskit/render_canvas_factory.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:meta/meta.dart';
+
+import '../../engine.dart';
+
+/// Caches canvases used to overlay platform views.
+class RenderCanvasFactory {
```

Comment:
Do we need the cache any more? "bitmaprenderer" canvases seem to be "free" in that they don't have their own bitmap memory. Instead, they just take ownership of the `ImageBitmap` memory that's given to them. Maybe what we can do is remember the last "composition" of render canvases and platform views (i.e. the sequence `[R1, V1, R2, V2, ..., Rk, Vk]`) and then optimistically render into them until we find that they need to be recomposited, which happens rarely (when platform views appear/disappear, or when the order changes). If/when we see that a recomposition is needed, we can then compute the minimum necessary "edit" to the sequence (we do that in the HTML renderer and in semantics), fix it, and then render into the new one.

Reply:
I agree, right now this factory basically just holds the base canvas and the `pictureToImageSurface`, the caching doesn't help us much. I'll remove this class in my forthcoming PR which reduces the number of overlays required.